src/AppInstallerRepositoryCore/CompositeSource.cpp

```diff
@@ -532,7 +578,21 @@ namespace AppInstaller::Repository
         // Search available sources
         for (const auto& source : m_availableSources)
         {
-            auto availableResult = source->Search(request);
+            SearchResult availableResult;
+
+            try
+            {
+                availableResult = source->Search(request);
+            }
+            catch (...)
+            {
+                LOG_CAUGHT_EXCEPTION();
+                AICLI_LOG(Repo, Warning, << "Failed to search source: " << source->GetDetails().Name);
+                result.AddFailureIfSourceNotPresent({ source, std::current_exception() });
+            }
+
+            // Move failures into the single result
+            std::move(availableResult.Failures.begin(), availableResult.Failures.end(), std::back_inserter(result.Failures));

```

Comment:
>std::move [](http://example.com/codeflow?start=12&length=9)

Should this use AddFailureIfSourceNotPresent to check duplicates?

Reply:
Probably, although I put that in mostly as a guard against unbounded correlation failures.